app/main.py

```diff
@@ -2,5 +2,16 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cash = {}
+
+    def inner(*args) -> int:
+        if cash.get(f"({func},{args})") is None:
+            result = func(*args)
+            cash[f"({func},{args})"] = result
+            print("Calculating new result")
+            return result
+
+        print("Getting from cache")
+        return cash.get(f"({func},{args})")
```

Comment:
You could avoid multiple conversions to a string of the function with its arguments, allocate a variable at the beginning of the inner logic:
```suggestion
        cached_key = (func, args)
        if cash.get(cached_key) is None:
            result = func(*args)
            cash[cached_key] = result
            print("Calculating new result")
            return result

        print("Getting from cache")
        return cash.get(cached_key)
```